src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/accession/repositories/JpaCustomMethodsRepositoryImpl.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.accession.repositories;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.jpa.repository.support.JpaEntityInformation;
+import org.springframework.data.jpa.repository.support.JpaEntityInformationSupport;
+import org.springframework.transaction.annotation.Transactional;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.ICustomMethodsRepository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import java.util.Set;
+
+public abstract class JpaCustomMethodsRepositoryImpl<ENTITY> implements ICustomMethodsRepository {
```

Comment:
I find these names too ambiguous, maybe something like `IAccessionedObjectCustomRepository` and `JpaAccessionedObjectCustomRepositoryImpl`? At least that would make clear they still apply to accessioned objects and not something else.